clang/include/clang/Sema/TemplateDeduction.h

```diff
@@ -33,6 +33,7 @@ namespace clang {
 class Decl;
 struct DeducedPack;
 class Sema;
+enum class TemplateDeductionResult;
```

Comment:
should we specify the underlying type in all of these?  I wonder if we can just specify it as `uint8_t`?  Would mean not having to store it as a bitfield if we didn't want.

Reply:
This is far from the first time we do unscoped enum → scoped enum refactoring so that we can forward declare enums. Me and @AaronBallman decided to stick to default `int` for scoped enums that are stored in bit-fieds. (This one is stored in bit-fields of width 8).